helix-view/src/document.rs

```diff
@@ -918,9 +962,22 @@ impl Document {
 
     /// Tab size in columns.
     pub fn tab_width(&self) -> usize {
-        self.language_config()
-            .and_then(|config| config.indent.as_ref())
-            .map_or(4, |config| config.tab_width) // fallback to 4 columns
+        if let Some(width) = self.tab_width_override {
+            width
+        } else {
+            self.language_config()
+                .and_then(|config| config.indent.as_ref())
+                .map_or(4, |config| config.tab_width) // fallback to 4 columns
+        }
+    }
+
+    /// Sets the tab size in columns.
+    ///
+    /// If `None` is passed, then [Document::tab_width]
+    /// will return the value specified by the language configuration
+    /// for this document.
+    pub fn set_tab_width(&mut self, tab_width_override: Option<usize>) {
+        self.tab_width_override = tab_width_override;
     }
```

Comment:
it looks like this is unused?

Reply:
I'd prefer removing it for now until we'd find a use-case for it. There's room for a follow-up to this PR that sets a document's tab width to supersede https://github.com/helix-editor/helix/pull/561 and for something like that I expect that we'll want to pass the tab_width as a usize rather than an `Option<usize>`. But in general I think it's better to lazily add those sorts of things so they can be guided by the need